Automattic/jetpack
# Social: Require a user connection to use the plugin.

If only a site connection is present (for example, when Jetpack Boost has been set up), then it was causing errors as the editor was loaded. This was down to us trying to retrieve and test the status of, social media connections for a non-existent user.

#### Changes proposed in this Pull Request:

This change adds some extra logic to ensure that we have a user connection present before we load the editor extension or the main admin page.

#### Other information:


#### Jetpack product discussion
p1664310513277449/1664282377.516619-slack-CBG1CP4EN

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:






* Install and set up Jetpack Boost
* Activate the Social plugin
* Load the editor or view the Social admin page

Without this change you'll see the connected admin page, and in the editor the plugin sidebar will be present. You'll see errors about trying to access array offsets on null.

With this change, the Social plugin will still disploy the 'Get Started' screen on the admin page, and not load the editor extension.

Additionally check that we don't display the Publicize form in the classic editor.
